pcr_profile.go

```diff
@@ -179,28 +144,109 @@ func (p *PCRProtectionProfile) AddProfileOR(profiles ...*PCRProtectionProfile) *
 	return p
 }
 
-// computePCRValues computes a list of different PCR value combinations from this PCRProtectionProfile.
-func (p *PCRProtectionProfile) computePCRValues(tpm *tpm2.TPMContext, values pcrValuesList) (pcrValuesList, error) {
-	if len(values) == 0 {
-		values = append(values, make(tpm2.PCRValues))
+// pcrProtectionProfileIterator provides a mechanism to perform a depth first traversal of instructions in a PCRProtectionProfile.
+type pcrProtectionProfileIterator struct {
+	instrs [][]pcrProtectionProfileInstr
+}
+
+func (iter *pcrProtectionProfileIterator) descendInToProfiles(profiles ...*PCRProtectionProfile) {
+	instrs := make([][]pcrProtectionProfileInstr, 0, len(profiles)+len(iter.instrs))
+	for _, p := range profiles {
+		pi := make([]pcrProtectionProfileInstr, len(p.instrs))
+		copy(pi, p.instrs)
```

Comment:
I'm not sure I understand the need of the copy,  pi is a slice which in itself is by value?

Reply:
Yeah, the copy is unnecessary in this case.